libs/providers/flagd/src/lib/service/in-process/grpc/grpc-fetch.ts

```diff
@@ -62,27 +65,33 @@ export class GrpcFetch implements DataFetch {
       this._logger?.debug('Received sync payload');
       dataFillCallback(data.flagConfiguration);
       connectCallback();
+      changedCallback([]); // flags changed list not supported
       resolveConnect();
       this._nextBackoff = initBackOffMs;
     });
 
-    this._syncStream.on('error', (err: Error) => {
-      this._logger?.error('Connection error, attempting to reconnect', err);
-      disconnectCallback();
-      this.reconnectWithBackoff(resolveConnect, dataFillCallback, connectCallback, disconnectCallback);
-    });
-
-    this._syncStream.on('end', () => {
-      this._logger?.error('Stream ended, attempting to reconnect');
-      disconnectCallback();
-      this.reconnectWithBackoff(resolveConnect, dataFillCallback, connectCallback, disconnectCallback);
+    this._syncStream.on('error', (err: ServiceError | undefined) => {
+      if (err?.code === status.CANCELLED) {
+        this._logger?.debug('Stream cancelled, will not be re-established');
```

Comment:
When we call stream.cancel in JS, it fires a an error, which is caught by the on error handler, so we get in an infinite reconnect loop event when we try to exit.